Classes/Player/AssetBuilder.swift

```diff
@@ -11,15 +11,7 @@ import AVFoundation
 
 class AssetBuilder {
     
-    let mediaEntry: MediaEntry
-    var assetHandler: AssetHandler?
-    
-    init(mediaEntry: MediaEntry) {
-        self.mediaEntry = mediaEntry
-    }
-
-    func getPreferredMediaSource() -> (MediaSource, AssetHandler.Type)? {
-        
+    static func getPreferredMediaSource(fromMediaEntry mediaEntry: MediaEntry) -> (MediaSource, AssetHandler.Type)? {
```

Comment:
ok, but 'from' is more appropriate here because you get from something and not for something